cluster-autoscaler/processors/nodeinfosprovider/mixed_nodeinfos_processor.go

```diff
@@ -36,17 +36,31 @@ import (
 
 const stabilizationDelay = 1 * time.Minute
 
+type cacheItem struct {
+	nodeInfo *schedulerframework.NodeInfo
+	added    time.Time
+}
+
 // MixedTemplateNodeInfoProvider build nodeInfos from the cluster's nodes and node groups.
 type MixedTemplateNodeInfoProvider struct {
-	nodeInfoCache map[string]*schedulerframework.NodeInfo
+	nodeInfoCache map[string]cacheItem
+	ttl           *time.Duration
 }
 
 // NewMixedTemplateNodeInfoProvider returns a NodeInfoProvider processor building
 // NodeInfos from real-world nodes when available, otherwise from node groups templates.
-func NewMixedTemplateNodeInfoProvider() *MixedTemplateNodeInfoProvider {
+func NewMixedTemplateNodeInfoProvider(ttl *time.Duration) *MixedTemplateNodeInfoProvider {
 	return &MixedTemplateNodeInfoProvider{
-		nodeInfoCache: make(map[string]*schedulerframework.NodeInfo),
+		nodeInfoCache: make(map[string]cacheItem),
+		ttl:           ttl,
+	}
+}
+
+func (p *MixedTemplateNodeInfoProvider) isCacheItemExpired(added time.Time) bool {
+	if p.ttl == nil || *p.ttl < 0 {
```

Comment:
Why do we need both nil duration and negative duration to achieve the same thing as setting the TTL to an arbitrarily large value?